src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -136,6 +144,12 @@ public AddressBook load() throws StorageOperationException {
         // create empty file if not found
         } catch (FileNotFoundException fnfe) {
             final AddressBook empty = new AddressBook();
+            // XXX hack to create an empty file since we now do not allow non existent files
+            try {
+                Files.createFile(path);
+            } catch (IOException e) {
+                throw new StorageOperationException("Cannot create file!");
+            }
```

Comment:
@LiHaoTan yup, you got it right ^^ you can still create file when you detect there is no file at the beginning though ^^